src/index.ts

```diff
@@ -0,0 +1,84 @@
+import { PreCompiler } from "gherking";
+import { Examples, ScenarioOutline, TableCell, TableRow } from "gherkin-ast"
+import { ScenarioOutlineNumberingConfig } from './types'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const debug = require("debug")("gpc:scenario-outline-numbering");
+
+const DEFAULT_CONFIG: ScenarioOutlineNumberingConfig = {
+    addParameters: false,
+    parameterDelimiter: ',',
+    parameterFormat: '${name} - ${parameters}',
+    addNumbering: true,
+    numberingFormat: '${i} - ${name}',
+    strictNaming: false
+};
+
+const NUMBERING_COLUMN = 'num';
+
+class ScenarioOutlineNumbering implements PreCompiler {
+    private config: ScenarioOutlineNumberingConfig;
+
+    constructor(config?: ScenarioOutlineNumberingConfig) {
+        this.config = {
+            ...DEFAULT_CONFIG,
+            ...(config || {}),
+        };
```

Comment:
I'd suggest checking if such properties as `parameterFormat` and `numberingFormat` are set, as they have a significant role later. Even though, we use the `DEFAULT_CONFIG`, with `config = { parameterFormat: "" }`, we can easily make all scenario outline nameless.